src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -597,17 +597,15 @@ void refresh() {
       if (territoryLastEntered != null) {
         final TerritoryAttachment ta = TerritoryAttachment.get(territoryLastEntered);
         if (ta != null) {
-          final Iterator<TerritoryEffect> iter = ta.getTerritoryEffect().iterator();
-          if (iter.hasNext()) {
+          final List<TerritoryEffect> territoryEffect = ta.getTerritoryEffect();
+          if (!territoryEffect.isEmpty()) {
             buf.append(" (");
           }
-          while (iter.hasNext()) {
-            buf.append(iter.next().getName());
-            if (iter.hasNext()) {
-              buf.append(", ");
-            } else {
-              buf.append(")");
-            }
+          buf.append(territoryEffect.stream()
+              .map(TerritoryEffect::getName)
+              .collect(Collectors.joining(", ")));
+          if (!territoryEffect.isEmpty()) {
```

Comment:
Nit: It might be easier to just do the empty check once:

```java
if (!territoryEffect.isEmpty()) {
  buf.append(" (");
  buf.append(territoryEffect.stream()
      .map(TerritoryEffect::getName)
      .collect(Collectors.joining(", ")));
  buf.append(")");
}
```

Reply:
Note to myself:
Use the overload of Collectors.join where prefix and suffix can be specified.